### PR TITLE
Splitting is necessary for assemblies with long stretches of N becaus…

### PR DESCRIPTION
**Pull request by TorHou**
…e MAFFT gets problems otherwise.

### CODE REVIEW

File: scripts/split_identify_fasta.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from argparse import ArgumentParser
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py:3]: I would add a shebang before this

`#!/usr/bin/env python3`

File: scripts/split_identify_fasta.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from Bio import SeqIO
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py]: I'm not a fan of biopython, as it's painfully slow and usually isn't the best solution in terms of memory allocation. 

I would recommend the following instead: https://pypi.org/project/pyfaidx/

For similar operations, I really like:
https://pypi.org/project/pyfastx/
https://github.com/lgautier/fastq-and-furious

But I mostly avoid biopython if I can....

File: scripts/split_identify_fasta.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            outf.write(">" + args.prefix + "_" + read.id + "_" + str(partnr) + "\n")
+            outf.write(part + "\n")
+
+            partnr += 1
+            start = nstop
+        part = s[start:]
+        outf.write(">" + args.prefix + "_" + read.id + "_" + str(partnr) + "\n")
+        outf.write(part + "\n")
+        
+        print("found " + str(partnr) + " parts")
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py]: You might want to capitalize "Found" and specifically put in a line-break here.

File: suggestCommands.pl [hunk 4]
```diff
@@ -122,11 +122,15 @@
 
 print qq(
 
+# If your contigs are interspersed with long stretches of unknown nucleotides (NNN...NNN) you should split them:
+# needs: Python 3 and biopython
```
Comment (evanbiederstedt) [inline suggestCommands.pl]: I would add a `requirements.txt` file for the python script. 

We should change the README as well.

**Review by evanbiederstedt — COMMENTED** [review]
We should add some directions in the README as well for this situation, and for users to install the desired libraries. 

I have some comments which should make things more efficient---I could dig into the script with you more actively if requested. 


If Alex thinks `*py` is an eyesore, someone could volunteer to convert to Perl ;)

File: scripts/split_identify_fasta.py (hunk 3 above)
Comment (TorHou) [reply scripts/split_identify_fasta.py]: output has been changed. Each fasta record while get one line... no more no less.

File: scripts/split_identify_fasta.py (hunk 1 above)
Comment (evanbiederstedt) [reply scripts/split_identify_fasta.py:3]: Thanks for the shebang :)

**Review by evanbiederstedt — APPROVED** [review]
LGTM!

It looks like you removed Biopython entirely, which avoids the dependency issues. 

Personally, I don't mind the dependencies! As I mentioned, I think there are some nice python libraries which would speed things up. But I salute your willingness to write you own parser :)

**Comment by evanbiederstedt** [comment]
Whenever you're ready @TorHou, please do feel free to merge the pull request.

**Comment by TorHou** [comment]
thanks for the review @evanbiederstedt

**Pull request merged**